File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: Optional[TreeNode]) -> int:
        res = float("inf")
        
        def dfs(node, count):
            nonlocal res
            if not node:
                return
            
            if (not node.left and not node.right):
                res = min(res, count)
                
            dfs(node.left, count + 1)
            dfs(node.right, count + 1)
            
        dfs(root, 1)
        
        return res if res != float("inf") else 0

File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
    var res = Infinity;
    
    const dfs = (node, count) => {
        if (!node){
            return;   
        }
        
        if (!node.left && !node.right){
            res = Math.min(res, count);
        }
        
        dfs(node.left, count + 1);
        dfs(node.right, count + 1);
    }
    
    dfs(root, 1);
    
    return (res === Infinity) ? 0 : res;
};